### PR TITLE
GHC 8.6

### PR DESCRIPTION
**Pull request by domenkozar**
If someone wants to pick this up, I hoped it would be a shorter rabbit hole :)

### CODE REVIEW

**Comment by jwiegley** [comment]
@domenkozar What's left to do?

**Comment by jwiegley** [comment]
With today's `unstable` I get:

```
error: attribute 'Glob_0_9_3' missing, at /Users/johnw/src/nix/nixpkgs/pkgs/development/haskell-modules/configuration-ghc-8.6.x.nix:52:10
```

**Comment by domenkozar** [comment]
I used stack to test this:

```yaml
resolver: nightly-2018-10-29

packages:
- .

extra-deps:
- serialise-0.2.1.0
- cborg-0.2.1.0
- cryptohash-sha256-0.11.101.0
- hashing-0.1.0.1
- monadlist-0.0.2
- repline-0.2.0.0
```
And the errors seem to be related to repl:

```haskell
/home/ielectric/dev/hnix/main/Repl.hs:183:5: error:
    * Couldn't match expected type `StateT IState m ()'
                  with actual type `HaskelineT (StateT IState m1) a0
                                    -> StateT IState m1 ()'
    * Probable cause: `evalRepl' is applied to too few arguments
      In the second argument of `($)', namely
        `evalRepl "hnix> " cmd options completer pre'
      In the expression:
        flip evalStateT initState
          $ evalRepl "hnix> " cmd options completer pre
      In an equation for `shell':
          shell pre
            = flip evalStateT initState
                $ evalRepl "hnix> " cmd options completer pre
    * Relevant bindings include
        pre :: Repl e m a (bound at main/Repl.hs:182:7)
        shell :: Repl e m a -> m () (bound at main/Repl.hs:182:1)
    |                                   
183 |     evalRepl "hnix> " cmd options completer pre
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

/home/ielectric/dev/hnix/main/Repl.hs:183:35: error:
    * Couldn't match expected type `Maybe Char'
                  with actual type `CompleterStyle (StateT IState m0)'
    * In the fourth argument of `evalRepl', namely `completer'
      In the second argument of `($)', namely
        `evalRepl "hnix> " cmd options completer pre'
      In the expression:
        flip evalStateT initState
          $ evalRepl "hnix> " cmd options completer pre
    |
183 |     evalRepl "hnix> " cmd options completer pre
    |                                   ^^^^^^^^^

/home/ielectric/dev/hnix/main/Repl.hs:183:45: error:
    * Couldn't match type `HaskelineT (StateT IState m) a'
                     with `CompleterStyle (StateT IState m1)'
      Expected type: CompleterStyle (StateT IState m1)
        Actual type: Repl e m a
    * In the fifth argument of `evalRepl', namely `pre'
      In the second argument of `($)', namely
        `evalRepl "hnix> " cmd options completer pre'
      In the expression:
        flip evalStateT initState
          $ evalRepl "hnix> " cmd options completer pre
    * Relevant bindings include
        pre :: Repl e m a (bound at main/Repl.hs:182:7)
        shell :: Repl e m a -> m () (bound at main/Repl.hs:182:1)
    |
183 |     evalRepl "hnix> " cmd options completer pre
    |                                             ^^^
```

**Comment by domenkozar** [comment]
Got it to compile. Once https://github.com/haskell-nix/hnix/pull/380/files is ready and we have nixpkgs version with lts-13 as a base, we can merge this.

**Comment by domenkozar** [comment]
@jwiegley ready to be merged - supports GHC 8.4 and 8.6.

File: main/Repl.hs [hunk 1]
```diff
@@ -180,7 +181,13 @@ completer = Prefix (wordCompleter comp) defaultMatcher
 
 shell :: (MonadNix e m, MonadIO m, MonadException m) => Repl e m a -> m ()
 shell pre = flip evalStateT initState $
-    evalRepl "hnix> " cmd options completer pre
+#if MIN_VERSION_repline(0, 2, 0)
+    evalRepl (return prefix() cmd options Nothing completer pre
```
Comment (mpickering) [inline main/Repl.hs:185]: This line causes the build to fail with a syntax error.

Comment (domenkozar) [reply main/Repl.hs:185]: Yes, sorry. I've fixed it in https://github.com/haskell-nix/hnix/pull/409

**Pull request merged**